app/plugins/Files/js/lifecycle.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+// Keeps track of currently existing files
+var files = {};
+
+// Make or update file
+function updateFile(file) {
+	console.log(file);
+	var nick = file.Nickname;
+	files[nick] = file;
+
+	// if updating or adding
+	var adding = eID(nick) ? false : true;
+
+	// Create only new ones
+	var fileElement = adding ? eID('filebp').cloneNode(true) : eID(nick);
+	fileElement.id = nick;
+
+	// DOM shortcut
+	// TODO: Don't know if bad practice because memleak or if it GCs well
+	var field = function(selector) {
+		return fileElement.querySelector(selector);
+	};
+
+	// Set field display values
+	field('.name').innerHTML = nick.length < 30 ? nick : nick.substr(0, 27) + '...';
+	field('.size').innerHTML = formatBytes(file.Filesize);
+	if (file.UploadProgress === 0) {
+		field('.time').innerHTML = 'Processing...';
+	} else if (file.UploadProgress < 100) {
+		field('.time').innerHTML = file.UploadProgress.toFixed(2) + '%';
+	} else {
+		field('.time').innerHTML = file.TimeRemaining + ' Blocks Remaining';
+	}
+
+	// Set graphic
+	field('.graphic i').classList.add('fa-file');
+
+	// Set availability graphic
+	if (file.Available) {
+		show(field('.yes'));
+		hide(field('.no'));
+	} else {
+		hide(field('.yes'));
+		show(field('.no'));
+	}
+
+	// Display file and add listeners if this was new
+	if (adding) {
+		eID('file-browser').appendChild(fileElement);
+		show(fileElement);
+	
+		// Give the file buttons clickability
+		field('.download').onclick = function() {
+			download(nick);
+		};
+		field('.share').onclick = function() {
+			share(nick);
+		};
+		field('.delete').onclick = function() {
+			deleteFile(nick);
+		};
+	}
+}
+
+// Regularly update the file library and status
+function update() {
+	IPC.sendToHost('api-call', '/renter/files/list', 'update-list');
+	IPC.sendToHost('api-call', '/renter/status', 'update-status');
+	
+	updating = setTimeout(update, 15000);
+}
```

Comment:
As I understand it, this calls `update` recursively after a timeout? Might you want to use `setInterval` instead?